CodeGeneration/Sources/generate-swift-syntax/GenerateSwiftSyntax.swift

```diff
@@ -154,6 +154,7 @@ struct GenerateSwiftSyntax: ParsableCommand {
       }
     )
 
+    let errorsLock = NSLock()
```

Comment:
Seems like we should use async/await with multiple tasks that we reduce after here, but that's a larger change. For this IMO just rename `previouslyGenerateFilesLock` and use it for both.

Reply:
Agreed that that should be the long term direction. Is this allowed in this project? I don't think that change should take long to make, provided the CI can verify that it works on Linux and windows.

Reply:
We require being able to compile with >= 5.7 currently and this is just a generation tool, so unless there's some bugs preventing it, I don't see any reason it would be an issue.

Reply:
[Here is an async version of this pr](https://github.com/apple/swift-syntax/compare/main...benpious:swift-syntax:switch-to-async). If this looks good to you I can abandon this pr and pr the Structured Concurrency version instead.

Reply:
LGTM, happy for you to put that up instead. Regarding the grouping below, I wouldn't worry about indenting personally - this is just for generation which typically shouldn't fail. What you could do is just separate them by file path, eg. ```
```
Errors occurred during generation.

'<file>':
<error here>
<empty newline>
'<file>':
<error here>
```

Reply:
Created https://github.com/apple/swift-syntax/pull/2245. It also incorporates this feedback about printing which file produced which error.

Reply:
Nice, using structured concurrency is so much cleaner. Thanks for putting it up. This means we can close this PR, right?